safe_transaction_service/history/serializers.py

```diff
@@ -406,14 +410,18 @@ def validate_delegator_signature(
     ) -> bool:
         ethereum_client = EthereumClientProvider()
         chain_id = ethereum_client.get_chain_id()
+
         # Accept a message with the current topt and the previous totp (to prevent replay attacks)
-        for previous_totp in (True, False):
+        for previous_totp, chain_id in list(
+            itertools.product((True, False), (chain_id, None))
+        ):
             message_hash = DelegateSignatureHelperV2.calculate_hash(
-                delegate, chain_id, previous_totp=previous_totp
+                delegate, chain_id=chain_id, previous_totp=previous_totp
             )
             safe_signatures = SafeSignature.parse_signature(signature, message_hash)
             if not safe_signatures:
-                raise ValidationError("Signature is not valid")
```

Comment:
Why this change? IMO, If the request doesn't have signatures will be right to raise an exception

`SafeSignature.parse_signature` would return a safe_signature in both cases, with our without `chain_id`

Reply:
> Why this change? IMO, If the request doesn't have signatures will be right to raise an exception
> 
> `SafeSignature.parse_signature` would return a safe_signature in both cases, with our without `chain_id`

Good point. I mistakenly assumed that `parse_signature` would not return any signature if the signature was not for the `message_has`. That validation happens further down in the control flow. Will fix 👍

Reply:
Changed in [cc92b7a](https://github.com/safe-global/safe-transaction-service/pull/2116/commits/cc92b7a1fc3848b5b36193d53c91dbbe4e7bc295)